compiler/rustc_query_system/src/query/plumbing.rs

```diff
@@ -44,6 +44,16 @@ enum QueryResult {
     Poisoned,
 }
 
+impl QueryResult {
+    /// Unwraps the query job and if poisoned will raise a [`FatalError`]
+    fn unwrap(self) -> QueryJob {
```

Comment:
This perhaps should be called `expect_job` or something?

Reply:
Yeah that would probably be better.